### DISCUSSION TITLE
Need help with Spacing Support in block.json

### DISCUSSION THREAD

**Discussion by realthemes** [WordPress/gutenberg#53603] [Q&A] [answered]
Hello, I'm stuck out of the blue again, my block supports `spacing support` in `block.json`, but when I specify `default values` ​​in `attributes`, they simply are not applied, I'm sure that I did everything as the documentation says, nevertheless nothing comes out, `padding-top` does not apply for my block.

My "supports":
```
"supports": {
	"spacing": {
		"margin": [ "top", "bottom" ],
		"padding": true,
		"__experimentalDefaultControls": {
			"margin": false,
			"padding": false
		}
	},
	"html": false
},
```

My "attributes":
```
"attributes": {
	"style": {
		"type":"object",
		"padding": {
			"top": "64px"
		}
	}
},
```

Expected to see padding-top 64px by default in my block as promised in [documentation](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-supports/#spacing).

**Comment by stokesman** [comment, marked as answer]
That bit of documentation isn't instructing how to set defaults but merely giving an (apparently incorrect) example of what the attributes will look like once they been customized by the user. To use them as the default they have to be under the `default` property of your attribute. Then spacing settings have to be under the `spacing` property (this is left out in the doc’s example). Like so:
```json
    "attributes": {
      "style": {
        "type": "object",
        "default": {
          "spacing":{"padding":{"top":"64px"}}
        }
      }
    }
```
I figured this out by setting padding on an example block and then switching to the Code editor to see how the attributes were output. For example, here’s how it looks using presets:
```
<!-- wp:paragraph {"style":{"spacing":{"padding":{"top":"var:preset|spacing|40"}}}} -->
```
It's good to keep in mind in case you need to figure out how other block supports apply any attribute values.

**Reply by realthemes** [reply]
@stokesman Thank you very much for your reply and good advice.
For those who will use this code, it is also required to set `__experimentalDefaultControls": { "padding": true }` if it was `false`
Like this:
```
"attributes": {
    "style": {
        "type": "object",
        "default": {
            "spacing": {
                "padding": {
                    "top": "64px"
                 }
             }
         }
    }
}
```
```
"supports": {
    "spacing": {
        "padding": true,
        "__experimentalDefaultControls": {
            "padding": true
        }
    }
}
```
Now it should work.

**Comment by spencerfinnell** [comment]
Note for anyone else looking to do this: as of now you need to use the computed value (i.e `64px`) of the preset and the editor will handle the conversion.

**Comment by realthemes** [comment]
@spencerfinnell 
I found an annoying bug related to these default styles, Here is the link https://github.com/WordPress/gutenberg/issues/56657
Unfortunately, my message to the developers is stuck in limbo, so I replaced the default styles with regular styles written in style.scss in all my projects.

**Reply by aut0poietic** [reply]
If anyone else encounters this bug but still wants to set defaults in the interim before the bug is patched, this seems to work:

In the `edit` method, you check to see if `attributes.style` or `attributes.style?.spacing?.padding` are undefined, and if so, restore defaults. Example:

```js
import blockAttributes from './attributes';
export default function Edit({ attributes, setAttributes }) {
	if (!attributes.style) {
		setAttributes({ style: blockAttributes.style.default });
	}
	if(!attributes.style?.spacing?.padding){
		setAttributes({ style: { ...attributes.style, spacing: blockAttributes.style.default.spacing } });
	}
}
```
Not a perfect solution, but should work for now, and shouldn't break anything if the bug's patched, right?

**Reply by philzest** [reply]
thanks for these response it is very helpful
